src/AlertsManagement/AlertsManagement/AlertCommands/GetAzureAlert.cs

```diff
@@ -0,0 +1,340 @@
+        public string State { get; set; }
+
+        /// <summary>
+        /// Alert Rule Id
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   ParameterSetName = AlertsListByFilterParameterSet,
+                   HelpMessage = "Filter on Alert Rule Id")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = AlertsListByTargetResourceGroupFilterParameterSet,
+                   HelpMessage = "Filter on Alert Rule Id")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = AlertsListByTargetResourceIdFilterParameterSet,
+                   HelpMessage = "Filter on Alert Rule Id")]
+        public string AlertRuleId { get; set; }
+
+        /// <summary>
+        /// Smart Group Id
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   ParameterSetName = AlertsListByFilterParameterSet,
+                   HelpMessage = "Filter all the alerts having the Smart Group Id")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = AlertsListByTargetResourceGroupFilterParameterSet,
+                   HelpMessage = "Filter all the alerts having the Smart Group Id")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = AlertsListByTargetResourceIdFilterParameterSet,
+                   HelpMessage = "Filter all the alerts having the Smart Group Id")]
+        public string SmartGroupId { get; set; }
+
+        /// <summary>
+        /// Include Context
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   ParameterSetName = AlertsListByFilterParameterSet,
+                   HelpMessage = "Include context (custom payload) of alert - true/false")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = AlertsListByTargetResourceGroupFilterParameterSet,
+                   HelpMessage = "Include context (custom payload) of alert - true/false")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = AlertsListByTargetResourceIdFilterParameterSet,
+                   HelpMessage = "Include context (custom payload) of alert - true/false")]
+        public bool IncludeContext { get; set; }
+
+        /// <summary>
+        /// Include EgressConfig
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   ParameterSetName = AlertsListByFilterParameterSet,
+                   HelpMessage = "Include EgressConfig - true/false")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = AlertsListByTargetResourceIdFilterParameterSet,
+                   HelpMessage = "Include EgressConfig - true/false")]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = AlertsListByTargetResourceGroupFilterParameterSet,
+                   HelpMessage = "Include EgressConfig - true/false")]
+        public bool IncludeEgressConfig { get; set; }
```

Comment:
See above note on parameter attributes - if a parameter is in *all* parameter sets, you should only have on parameter attribute, and it should not name a parameter set.

Reply:
As mentioned above - what about InputObject parameter set?

Reply:
This needs to be updated for GA